added testing deeplinks (#9)
ScreaMy7/owasp-mastg: demos/android/MASVS-CODE/MASTG-DEMO-0062/MASTG-DEMO-0062.md

---
platform: android
title: Unvalidated URL from Deep Link Loaded in WebView with semgrep
id: MASTG-DEMO-0062
code: [kotlin]
test: MASTG-TEST-0288
status: new
---

### Sample

The following is a sample code file that contains a function to handle a deep link, which insecurely loads a URL into a WebView.

> {{ MastgTest_reversed.java }}

Reviewer: ```suggestion
{{ MastgTest.kt # MastgTest_reversed.java }}
```

### Steps

Let's run @MASTG-TOOL-0110 rules against the sample code.

{{ ../../../../rules/mastg-android-unvalidated-deeplink-data.yml }}

{{ run.sh }}

### Observation

> The rule has identified a dangerous data flow from a source `getQueryParameter` to a sink `loadUrl`.

Reviewer: ```suggestion
The output file shows usage of dangerous data flow from a source `getQueryParameter` to a sink `loadUrl`.
```
